### PR TITLE
Added ability to disable specific metatags.

### LINKED ISSUE #20: Allow enabling individual metatags site-wide
When configuring metatags for various instances (global, node, content types, etc.), I'm having to scroll past lots of metatags I've never going to use on my site (e.g. Audio URL, Video, etc.).

Can we add a master list of all metatags to the settings area with a checkbox next to each one to enable/disable them on a site-wide basis? That'd really help with editing metatags since you're then only seeing the subset that you'll most likely want to use...

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/backdrop-contrib/metatag/issues/20

### CODE REVIEW

File: metatag.admin.inc [hunk 1]
```diff
@@ -766,6 +766,29 @@ function metatag_admin_settings_form() {
     '#default_value' => $settings->get('pager_string'),
   );
 
+  $form['disabled'] = array(
```
Comment (jenlampton) [inline metatag.admin.inc]: maybe enabled?

File: config/metatag.settings.json [hunk 2]
```diff
@@ -17,5 +17,91 @@
     "metatag_enable_node__post": 1,
     "metatag_enable_taxonomy_term": 1,
     "metatag_enable_taxonomy_term__tags": 1,
-    "metatag_enable_user": 1
+    "metatag_enable_user": 1,
+    "disabled": {
```
Comment (jenlampton) [inline config/metatag.settings.json]: maybe enabled?

File: metatag.admin.inc [hunk 3]
```diff
@@ -766,6 +766,29 @@ function metatag_admin_settings_form() {
...
+  $metatags = metatag_get_info(NULL, NULL, TRUE);
+  $options = array();
+  foreach ($metatags['tags'] as $tag) {
+    $options[$tag['name']] = t('@group_label: @tag_label', array(
+      '@group_label' => $metatags['groups'][$tag['group']]['label'],
+      '@tag_label' => $tag['label'],
+    ));
+  }
+  $default = $settings->get('disabled');
+  $form['disabled']['disabled'] = array(
```
Comment (jenlampton) [inline metatag.admin.inc]: maybe enabled enabled?

File: metatag.module [hunk 4]
```diff
@@ -1970,7 +1970,7 @@ function metatag_form_alter(&$form, $form_state, $form_id) {
  *   The meta tag name, e.g. description, for which the info shall be returned,
  *   or NULL to return an array with info about all meta tags.
  */
-function metatag_get_info($type = NULL, $name = NULL) {
+function metatag_get_info($type = NULL, $name = NULL, $include_disabled = FALSE) {
```
Comment (jenlampton) [inline metatag.module:1969]: $enabled = true?

File: metatag.module [hunk 5]
```diff
@@ -1993,13 +1993,19 @@ function metatag_get_info($type = NULL, $name = NULL) {
       $info = module_invoke_all('metatag_info');
       $info += array('tags' => array(), 'groups' => array());
 
-      // Merge in default values.
+      $disabled = config_get('metatag.settings', 'disabled');
```
Comment (jenlampton) [inline metatag.module]: $enabled?

File: metatag.module [hunk 6]
```diff
@@ -1993,13 +1993,19 @@ function metatag_get_info($type = NULL, $name = NULL) {
...
       foreach ($info['tags'] as $key => $data) {
+        // Merge in default values.
         $info['tags'][$key] += array(
           // Merge in default values.
           'name' => $key,
           'class' => 'BackdropTextMetaTag',
         );
+
+        // Remove disabled tags.
+        if (!empty($disabled[$key]) && !$include_disabled) {
```
Comment (jenlampton) [inline metatag.module]: if $enabled[$key]?

**Review by jenlampton — CHANGES_REQUESTED** [review]
oh jeez, did I never post this review? I'm so sorry :(

My thought process was that we should probably disable everything initially, then enable only the things we need, so maybe the variables should be named the other way around?

**Comment by ghost** [comment]
@jenlampton No worries! I think you somehow reviewed an old version of this PR. When I look at the file changes tab, I see things how you want them already... Can you confirm?

**Comment by jenlampton** [comment]
yep, you're right. Will give this another test :)

**Comment by jenlampton** [comment]
I re-reviewed this PR and it looks great. I think we need an update hook to enable all tags for existing sites (to recreate current behavior), since for new sites we want most of them disabled.

**Comment by ghost** [comment]
I can no longer push to this PR, so I made my own fork and did it there instead. Closing this PR in favour of https://github.com/backdrop-contrib/metatag/pull/66.

**Pull request closed without merging**